cmd/coordinated/http.go

```diff
@@ -21,10 +25,44 @@ type HTTP struct {
 // Serve runs an HTTP server on the specified local address. This serves
 // connections forever, and probably wants to be run in a goroutine. Panics on
 // any error in the initial setup or in accepting connections.
-func (h *HTTP) Serve() {
+func (h *HTTP) Serve(logRequests bool, logFormat string, logger *logrus.Logger) {
 	r := mux.NewRouter()
 	r.PathPrefix("/").Subrouter()
 	restserver.PopulateRouter(r, h.coord)
 	r.Handle("/metrics", promhttp.Handler())
-	http.ListenAndServe(h.laddr, r)
+
+	n := negroni.New()
+	n.Use(negroni.NewRecovery())
+
+	// Wrap the root handler in a logger if desired.
+	var handler http.Handler = r
+	if logRequests {
+		handler = logWrapper(logFormat, logger, handler)
+	}
+	n.UseHandler(handler)
+
+	http.ListenAndServe(h.laddr, n)
+}
+
+// logWrapper creates a wrapping logger for the given handler. It is setup this
+// way rather than conforming to the negroni paradigm because the API fo the
+// requestlog package, which this uses, is not directly compatible.
+func logWrapper(logFormat string, logger *logrus.Logger, inner http.Handler) http.Handler {
+	var reqLog requestlog.Logger
+	switch logFormat {
+	case "ncsa":
```

Comment:
Does this format deserve explanation?

Reply:
I was mirroring the names of the loggers in the requestlog package, but I agree I don't totally love this. https://godoc.org/github.com/google/go-cloud/requestlog For one thing, with a few searches I can't seem to find an actual definition for NCSA... I could call this "combined" but that feels only slightly better. Thoughts?

Reply:
With a presidence I'm happy the way things are here :)

Reply:
Though a link to that may still be useful.

Reply:
Link added!